### PR TITLE
Razor Components: Routing Preview3 updates

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #10717 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/routing?view=aspnetcore-3.0&branch=pr-en-us-11396)

Let's drop the explicit references to the common (Blazor) sample app since there are deltas between Blazor and RC right now.

### CODE REVIEW

File: aspnetcore/razor-components/routing.md [hunk 1]
```diff
@@ -14,21 +14,29 @@ By [Luke Latham](https://github.com/guardrex)
 
 Learn how to route requests in apps and about the NavLink component.
 
-[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/razor-components/common/samples/) ([how to download](xref:index#how-to-download-a-sample)). See the [Get started](xref:razor-components/get-started) topic for prerequisites.
```
Comment (tdykstra) [inline aspnetcore/razor-components/routing.md]: No more sample app?

File: aspnetcore/razor-components/routing.md [hunk 2]
```diff
@@ -14,21 +14,29 @@ By [Luke Latham](https://github.com/guardrex)
...
+## ASP.NET Core endpoint routing integration
+
+Razor Components are integrated into [ASP.NET Core routing](xref:fundamentals/routing). The app is configured to accept incoming connections for interactive Razor Components with `MapComponentHub<TComponent>` in `Startup.Configure`. `MapComponentHub` specifies that the root component `App` should be rendered within a DOM element matching the selector `app`:
```
Comment (tdykstra) [inline aspnetcore/razor-components/routing.md]: "The" app?  What app is this referring to - the sample app?

File: aspnetcore/razor-components/routing.md [hunk 3]
```diff
@@ -14,21 +14,29 @@ By [Luke Latham](https://github.com/guardrex)
...
-The `<Router>` component enables routing, and a route template is provided to each accessible component. The `<Router>` component appears in the *App.cshtml* file:
+The `<Router>` component enables routing, and a route template is provided to each accessible component. The `<Router>` component appears in the *Components/App.razor* file:
 
 ```cshtml
-<Router AppAssembly=typeof(Program).Assembly />
+<Router AppAssembly="typeof(Program).Assembly" />
 ```
 
-When a *\*.cshtml* file with an `@page` directive is compiled, the generated class is given a [RouteAttribute](/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders whichever component has a route template that matches the requested URL.
+When a *.razor* or *.cshtml* file with an `@page` directive is compiled, the generated class is given a <xref:Microsoft.AspNetCore.Mvc.RouteAttribute> specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders whichever component has a route template that matches the requested URL.
```
Comment (tdykstra) [inline aspnetcore/razor-components/routing.md]: 👍 for switching from "\*.extension" to just ".extension"

File: aspnetcore/razor-components/routing.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/razor-components/routing.md]: Not for now ... Blazor routing is different from the endpoint routing used by Razor Components, and the sample app is a Blazor app (by-design at this time, but subject to change by the time we reach RTM). For the time being, this topic will just describe the Razor Components approach. This was one of the things that Dan is guiding: I think he will want me to create a separate Blazor-based Routing topic for the Blazor node by the time we hit RTM.

**Comment by khenzar** [comment]
Is the new routing preventing any additional non-RC routes? 
I am using [Microsoft.AspNetCore.Authentication.AzureAD.UI](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.AzureAD.UI/3.0.0-preview3-19153-02) nuget package (version 3.0.0-preview3-19153-02) to enable AzureAD login and **it worked fine in Preview 2**.
The package sets up a 'virtual' controller called 'Account' in an Area called 'AzureAd' that allows the 'AzureAd/Account/SignIn' and 'AzureAd/Account/SignOut' links to work.
When updating to RC Preview 3, navigating to those links now fails with:
```
System.InvalidOperationException: 'Router' cannot find any component with a route for '/AzureAD/Account/SignIn', and no fallback is defined.
   at Microsoft.AspNetCore.Components.Routing.Router.Refresh()
...
```
Is this is known issue?

**Comment by guardrex** [comment]
Hello @khenzar ... Open this over on the aspnet/AspNetCore issues for the engineers to see. We only handle doc issues here.

https://github.com/aspnet/AspNetCore/issues

**Pull request merged**